api/metrics-api-jaxrs-1.1/src/main/java/org/hawkular/metrics/api/jaxrs/handler/GaugeHandler.java

```diff
@@ -247,9 +253,34 @@ public Response findGaugeData(
                     percentiles = new Percentiles(Collections.<Double>emptyList());
                 }
 
-                return metricsService
-                        .findGaugeStats(metricId, timeRange.getStart(), timeRange.getEnd(), buckets,
-                                percentiles.getPercentiles())
+                Observable<TimeRange> observableTimeRange = Observable.just(timeRange);
+
+                if (Boolean.TRUE.equals(fromEarliest)) {
```

Comment:
A few lines below there is this code part of the same if block:

```
return list.subList(index, list.size());
```

This if removes all empty buckets from start of the list.